echoshader/dev_class/echo_gram.py

```diff
@@ -0,0 +1,499 @@
+        Number of columns when viewing all grams
+        If the value is set to '1', there will be only one layout column
+        If there are three kinds grams(frequencies) and the value is set to '3', there will be only one layout Row (three Columns)
+
+    time_range_picker : panel.widgets
+        Picker panel widget to input datetime range
+        See more in : https://panel.holoviz.org/reference/widgets/DatetimeRangePicker.html#widgets-gallery-datetimerangepicker
+
+    datetime_range_input : panel.widgets
+        Input panel widget to input datetime range
+        See more in : https://panel.holoviz.org/reference/widgets/DatetimeRangeInput.html
+
+    channel_select : panel.widgets
+        Select panel widget to select frequency
+        See more in : https://panel.holoviz.org/reference/widgets/Select.html#widgets-gallery-select
+
+    color_map : panel.widgets
+        Text input panel widget to input colormap
+        See more in : https://panel.holoviz.org/reference/widgets/TextInput.html#widgets-gallery-textinput
+
+    range_clim : panel.widgets
+        Editable range slider widget to select clim range
+        See more in : https://panel.holoviz.org/reference/widgets/EditableRangeSlider.html#widgets-gallery-editablerangeslider
+
+    widgets : panel.widgets
+        Arrange multiple panel objects in a vertical container
+        See more in : https://panel.holoviz.org/reference/layouts/WidgetBox.html#layouts-gallery-widgetbox
+
+    gram : hvplot.image
+        Echogram
+        Only be accessed after calling method 'view_gram()'
+
+    box : holoviews.streams.BoundsXY
+        Bound values of selected box
+        Only be accessed after calling method 'view_gram()'
+
+    bounds : holoviews.streams.Bounds
+        Bounds plot in echogram
+        Only be accessed after calling method 'view_gram()'
+
+    all_gram : holoviews.NdLayout
+        Echograms with all frequencies
+        Only be accessed after calling method 'view_all_gram()'
+
+    Methods
+    -------
+    _sync_widget(self):
+        Initialize widgets attributes
+        Called in __init__(self, MVBS_ds, **params)
+
+    view_gram(self):
+        Get single echogram with control panel widgets
+
+    view_all_gram(self):
+        Get all echograms with control panel widgets
+
+    get_box_data(self):
+        Get MVBS data with a specific frequency in selected box
+
+    get_all_box_data(self):
+        Get MVBS data with all frequencies in selected box
+
+    Examples
+    --------
+        echogram = Echogram(MVBS_ds)
+
+        echogram.bound_opts=holoviews.opts(line_color='black')
+
+        panel.Row(echogram.widgets,echogram.view_gram)
+
+    """
+
+    def __init__(self, MVBS_ds, **params):
+        """
+        Constructs all the necessary attributes for the echogram object.
+
+        Parameters
+        ----------
+        MVBS_ds : xarray.Dataset
+            MVBS Dataset with coordinates 'ping_time', 'channel', 'echo_range'
+
```

Comment:
You should also add `**params` in parameters, since this is technically a valid input for this class.

Reply:
@lsetiawan  What is **params used for? Can you give me some examples?